modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/skin/Utils.java

```diff
@@ -213,8 +212,29 @@ public static int computeTruncationIndex(Font font, String text, double width) {
         return index;
     }
 
-    public static String computeClippedText(Font font, String text, double width,
-                                     OverrunStyle type, String ellipsisString) {
+    /**
+     * Computes the actual text to be shown in the Labeled with the text wrapping disabled:
+     * unmodified if it fits into available area,
+     * or with the ellipsis string inserted into strategic place(s) if it does not.
+     * The latter case will cause {@code textTruncated} reference set to {@code true} (the caller is expected
+     * to set the flag to {@code false} before invoking this method).
+     *
+     * @param font the font
+     * @param text the original text
+     * @param width the available width
+     * @param type the truncation style
+     * @param ellipsisString the ellipsis string
+     * @param textTruncated the reference to a flag indicating the truncation
+     * @return the actual text to be shown, with the ellipsis string inserted when needed
+     */
+    public static String computeClippedText(
+        Font font,
+        String text,
+        double width,
+        OverrunStyle type,
+        String ellipsisString,
+        AtomicBoolean textTruncated
```

Comment:
Instead of returning the clipped text with the return value, and the truncation flag with an argument, have you considered just using a record to return both at the same time? That would be a cleaner solution.

Reply:
Not really a cleaner solution: `computeClippedWrappedText()` might get called multiple times (for each line of text), and doing an extra allocation per call is less optimal than passing a reference to an AtomicBoolean.